frontend/src/services/actions.ts

```diff
@@ -57,8 +57,8 @@ const messageActions = {
       store.dispatch(addAssistantMessage(message.args.thought));
     }
     if (
-      !message.args.is_confirmed ||
-      message.args.is_confirmed !== "rejected"
+      !message.args.confirmation_state ||
```

Comment:
Use a switch statement for `confirmation_state` checks.

Reply:
The presence check was redundant